db/sqlglue.c

```diff
@@ -5006,10 +4957,12 @@ int sqlite3BtreeCommit(Btree *pBt)
     if (clnt->selectv_arr)
         currangearr_coalesce(clnt->selectv_arr);
 
+    if (clnt->ctrl_sqlengine == SQLENG_NORMAL_PROCESS) {
```

Comment:
In addition, for standalone queries, it is possible that sqlite engine decides to abort and calls Rollback instead of commit; we should reset the flag as well.